xonsh/completers/python.py

```diff
@@ -95,10 +95,14 @@ def attr_complete(prefix, ctx, filter_func):
         except:  # pylint:disable=bare-except
             continue
         a = getattr(val, opt)
-        if callable(a):
-            rpl = opt + '('
-        elif isinstance(a, abc.Iterable):
-            rpl = opt + '['
+        if (builtins.__xonsh_env__['COMPLETIONS_BRACKETS'] and
+                        opt not in ['stdin', 'stdout', 'stderr']):
+            if callable(a):
+                rpl = opt + '('
+            elif isinstance(a, abc.Iterable):
```

Comment:
Actually, now that I think about it, this should `isinstance(a, (abc.Sequence, abc.Mapping))` and the special case file handles above should be removed.  We only want to add an `[` when there is a `__getitem__`.  As a counter example, this implementation will still add an `[` to sets, file handles, and other iterable but non-indexable  objects.